### PR TITLE
Add Checkbox and Radio Button Group Functionality

### PR DESCRIPTION
**Pull request by cirdes**
This pull request introduces the following changes:

**Checkbox Group**
- Added checkbox_group.rb to handle checkbox group logic.
- Added checkbox_group_controller.js to manage checkbox group interactions.

**Radio Button**
- Added radio_button.rb for radio button functionality.

**Form Enhancements**
- Updated form_field_controller.js and form_field_error.rb to support new checkbox and radio button features.

**Button Updates**
Modified button.rb to replace type instead of concatenate.

**Index Update**
- Imported and registered CheckboxGroupController in index.js

### CODE REVIEW

File: lib/rbui/checkbox/checkbox_group_controller.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ['checkbox'];
```
Comment (pierry01) [inline lib/rbui/checkbox/checkbox_group_controller.js:4]: `static targets = ["checkbox"];`

Comment (pierry01) [inline lib/rbui/checkbox/checkbox_group_controller.js:4]:
```suggestion
  static targets = ["checkbox"];
```

File: lib/rbui/checkbox/checkbox_group_controller.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  connect() {
+    if (this.element.hasAttribute("data-required")) {
+      this.checkboxTargets.forEach(checkbox => {
```
Comment (pierry01) [inline lib/rbui/checkbox/checkbox_group_controller.js:8]: can we use just the `required` attribute?

File: lib/rbui/checkbox/checkbox_group_controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        checkbox.required = true;
+      });
+    }
+  }
+
+  onChange(event) {
+    if (this.element.hasAttribute("data-required")) {
```
Comment (pierry01) [inline lib/rbui/checkbox/checkbox_group_controller.js:15]: can we use just the `required` attribute?

File: lib/rbui/button/button.rb [hunk 4]
```diff
@@ -2,7 +2,8 @@
 
 module RBUI
   class Button < Base
-    def initialize(variant: :primary, size: :md, icon: false, **attrs)
+    def initialize(type: :button, variant: :primary, size: :md, icon: false, **attrs)
+      @type = type
```
[the remark this answers is not in the document]
Comment (cirdes) [reply lib/rbui/button/button.rb:6]: @pierry01 , exactly! This should be able to fix that bug

File: lib/rbui/checkbox/checkbox_group_controller.js (hunk 3 above)
Comment (cirdes) [reply lib/rbui/checkbox/checkbox_group_controller.js:15]: no =/

File: lib/rbui/form/form_field_controller.js [hunk 5]
```diff
@@ -1,13 +1,15 @@
 import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
-  static targets = ['input', 'message']
+  static targets = ['input', 'error']
```
Comment (pierry01) [inline lib/rbui/form/form_field_controller.js:4]:
```suggestion
  static targets = ["input", "error"];
```

File: lib/rbui/form/form_field_controller.js [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
   static values = { shouldValidate: false }
 
 
   connect() {
-    if (this.messageTarget.textContent) {
+    if (this.errorTarget.textContent) {
       this.shouldValidateValue = true;
+    } else {
+      this.errorTarget.classList.add('hidden');
```
Comment (pierry01) [inline lib/rbui/form/form_field_controller.js:12]:
```suggestion
      this.errorTarget.classList.add("hidden");
```

File: lib/rbui/form/form_field_controller.js [hunk 7]
```diff
@@ -30,9 +32,11 @@ export default class extends Controller {
     if (!this.shouldValidateValue) return;
 
     if (this.inputTarget.validity.valid) {
-      this.messageTarget.textContent = '';
+      this.errorTarget.textContent = '';
+      this.errorTarget.classList.add('hidden');
```
Comment (pierry01) [inline lib/rbui/form/form_field_controller.js:36]:
```suggestion
      this.errorTarget.textContent = "";
      this.errorTarget.classList.add("hidden");
```

**Review by pierry01 — APPROVED** [review]

**Pull request merged**